exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/UnlimitedRawBatchBuffer.java

```diff
@@ -63,6 +64,15 @@ public RawFragmentBatch take() throws IOException, InterruptedException {
       return batch;
     }
 
+    @Override
+    public RawFragmentBatch poll(long timeout, TimeUnit timeUnit) throws InterruptedException, IOException {
+      RawFragmentBatch batch = buffer.poll(timeout, timeUnit);
```

Comment:
Does poll return null when timeout is reached? if not, then you should handle timeout exception.